alg/teca_mesh_padding.cxx

```diff
@@ -0,0 +1,283 @@
+    cerr << teca_parallel_id()
+        << "teca_mesh_padding::get_output_metadata" << endl;
+#endif
+    (void)port;
+
+    // add in the array we will generate
+    teca_metadata out_md(input_md[0]);
+
+    return out_md;
+}
+
+// --------------------------------------------------------------------------
+std::vector<teca_metadata> teca_mesh_padding::get_upstream_request(
+    unsigned int port,
+    const std::vector<teca_metadata> &input_md,
+    const teca_metadata &request)
+{
+#ifdef TECA_DEBUG
+    cerr << teca_parallel_id()
+        << "teca_mesh_padding::get_upstream_request" << endl;
+#endif
+    (void) port;
+    (void) input_md;
+
+    std::vector<teca_metadata> up_reqs;
+
+    // send up
+    up_reqs.push_back(request);
+    return up_reqs;
+}
+
+// --------------------------------------------------------------------------
+const_p_teca_dataset teca_mesh_padding::execute(
+    unsigned int port,
+    const std::vector<const_p_teca_dataset> &input_data,
+    const teca_metadata &request)
+{
+#ifdef TECA_DEBUG
+    cerr << teca_parallel_id() << "teca_mesh_padding::execute" << endl;
+#endif
+
+    (void)port;
+    (void)request;
+
+    // get the input
+    const_p_teca_cartesian_mesh in_mesh =
+        std::dynamic_pointer_cast<const teca_cartesian_mesh>(input_data[0]);
+
+    if (!in_mesh)
+    {
+        TECA_ERROR("empty input, or not a mesh")
+        return nullptr;
+    }
+
+    // create output and copy metadata, coordinates, etc
+    p_teca_cartesian_mesh out_mesh = teca_cartesian_mesh::New();
+
+    out_mesh->copy_metadata(
+        std::const_pointer_cast<teca_cartesian_mesh>(in_mesh));
+
+    teca_metadata &out_metadata = out_mesh->get_metadata();
+
+    size_t py_low = this->get_py_low();
+    size_t py_high = this->get_py_high();
+    size_t px_low = this->get_px_low();
+    size_t px_high = this->get_px_high();
+
+    // get the coordinate axes
+    std::string y_name;
+    std::string x_name;
+    in_mesh->get_y_coordinate_variable(y_name);
+    in_mesh->get_x_coordinate_variable(x_name);
+
+    const_p_teca_variant_array y_crds = in_mesh->get_y_coordinates();
+    const_p_teca_variant_array x_crds = in_mesh->get_x_coordinates();
+    const_p_teca_variant_array z_crds = in_mesh->get_z_coordinates();
+
+    size_t n_y = y_crds->size();
+    size_t n_x = x_crds->size();
+    size_t n_z = z_crds->size();
+
+    size_t ny_new = py_low + n_y + py_high;
+    size_t nx_new = px_low + n_x + px_high;
+
+    size_t nxy_new = nx_new * ny_new;
+
+
+    unsigned long req_whole_extent[6] = {0};
+    unsigned long req_extent[6] = {0};
+
+    req_whole_extent[1] = nx_new - 1;
+    req_whole_extent[3] = ny_new - 1;
+    req_whole_extent[5] = n_z;
+    req_extent[1] = nx_new - 1;
+    req_extent[3] = ny_new - 1;
+    req_extent[5] = n_z;
+
+    out_mesh->set_whole_extent(req_whole_extent);
+    out_mesh->set_extent(req_extent);
+
+    out_metadata.set("whole_extent", req_whole_extent, 6);
+    out_metadata.set("extent", req_extent, 6);
```

Comment:
this code is only correct if extent == whole_extent. you should instead direvctly query both extent and whole_extent from the input and make adjustments to the output based on the values found